lib/charms/operator_libs_linux/v0/snap.py

```diff
@@ -524,17 +525,33 @@ def __iter__(self) -> Iterable["Snap"]:
 
     def __getitem__(self, snap_name: str) -> Snap:
         """Return either the installed version or latest version for a given snap."""
+        if not self.snapd_installed:
+            raise SnapError("snapd is not installed or not in /usr/bin") from None
```

Comment:
Would this not be better off in `__init__`?

Reply:
Agree. We want to catch this as quickly as possible, correct?